src/components/EditAlbum.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import { addAlbum, editAlbum } from "@/actions";
+import { Album, genreEnum } from "@/db/schema";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import toast from "react-hot-toast";
+
+export default function EditAlbum({ album }: { album: Album }) {
+  const [title, setTitle] = useState(album.title);
+  const [artist, setArtist] = useState(album.artist);
+  const [genre, setGenre] = useState(album.genre);
+  const [button, setButton] = useState(true);
+
+  const genres = genreEnum.enumValues;
+  const router = useRouter();
+
+  async function handleAdd() {
+    setButton(!button);
+    if (title.length < 3 || artist.length < 3) {
+      toast.error(
+        "please enter valid album, inputs cannot be shorter than 3 lol",
+      );
+    } else {
+      try {
+        await editAlbum({ title, artist, genre });
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    router.refresh();
+  }
+
+  return (
+    <div>
+      <button
+        type="button"
+        onClick={handleAdd}
+        className="text-center m-3 p-3 rounded-full bg-blue-500 text-white hover:bg-blue-600"
+      >
+        {button ? "edit album" : "save changes"}
+      </button>
+      {button ? (
+        <p></p>
```

Comment:
displaying empty tags is always a bit weird. If you do not need it hear, invert the condition and return null at the end instead of returning an empty `p` tag